pkg/gofr/datasource/sql/sql_migrate.go

```diff
@@ -0,0 +1,172 @@
+package sql
+
+import (
+	"database/sql"
+	"fmt"
+	"reflect"
+	"regexp"
+	"strings"
+	"time"
+
+	"gofr.dev/pkg/gofr/logging"
+)
+
+// GenerateCreateTableSQL generates a SQL CREATE TABLE statement for the given struct.
+func GenerateCreateTableSQL(structType interface{}, dbType string, dropIfExists bool) (string, string, string, string, string, error) {
+	dbType = strings.ToUpper(strings.TrimSpace(dbType))
+	t := reflect.TypeOf(structType)
+	tableName := ToSnakeCase(t.Name())
+
+	fields := []string{}
+	indexes := []string{}
+	uniqueIndexes := []string{}
+	foreignKeys := []string{}
+	triggers := []string{}
+
+	for i := 0; i < t.NumField(); i++ {
+		field := t.Field(i)
+		columnName := ToSnakeCase(field.Name)
+
+		sqlType := ""
+		comment := ""
+		checkConstraint := ""
+		foreignKey := ""
+		sqlTags := field.Tag.Get("sql")
+		tagParts := strings.Split(sqlTags, ",")
+		for _, tag := range tagParts {
+			tag = strings.TrimSpace(tag)
+			if strings.HasPrefix(tag, "comment(") {
+				commentPattern := regexp.MustCompile(`comment\((.+)\)`)
+				matches := commentPattern.FindStringSubmatch(tag)
+				if len(matches) == 2 {
+					comment = matches[1]
+				}
+			} else if strings.HasPrefix(tag, "check(") {
+				checkPattern := regexp.MustCompile(`check\((.+)\)`)
+				matches := checkPattern.FindStringSubmatch(tag)
+				if len(matches) == 2 {
+					checkConstraint = matches[1]
+				}
+			} else if strings.HasPrefix(tag, "fk(") {
+				fkPattern := regexp.MustCompile(`fk\((.+):(.+)\)`)
```

Comment:
Please extract all regexp computing out of the method, or find a way to compute them only once

They are time consuming